### PR TITLE
dockeriseeee

### PR DESCRIPTION
**Pull request by Oceankoh**
me like docker but docker don't like me :(

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+RUN mkdir /var/www/ctfx
+
+COPY include /var/www/ctfx/include
+COPY install /var/www/ctfx/install
+COPY composer.* /var/www/ctfx/
+COPY writable /var/www/ctfx/writable
+
+RUN chmod -R 777 /var/www/ctfx/writable
+RUN chown -R www-data:www-data /var/www/ctfx/include/config
+
```
Comment (dbsqwerty) [inline Dockerfile:18]: is there no need to copy the db.default.inc.php and config.default.inc.php to db.inc.php and config.inc.php respectively. Perhaps we could either make use of sed to write the config set in docker-compose or to get the env vars set in docker-compose to apply to the config files

File: db/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# needed for intialization
+ENV MYSQL_ROOT_PASSWORD="seandontforgetmewhenyougetrichpls"
+
+COPY db.sql /docker-entrypoint-initdb.d/
+
+RUN ["/usr/local/bin/docker-entrypoint.sh", "mysqld", "--datadir", "/initialized-db"]
+
+FROM mysql:8.0
+
+COPY --from=builder /initialized-db /var/lib/mysql
```
Comment (dbsqwerty) [inline db/Dockerfile:15]: same as the other Dockerfile. Is there a way to incorporate env configs in the setting up of the database as well as the setting up of the admin user account

File: db/db.sql [hunk 3]
```diff
@@ -0,0 +1,498 @@
+CREATE DATABASE mellivora CHARACTER SET utf8 COLLATE utf8_general_ci;
+CREATE USER 'mellivora'@'%' IDENTIFIED BY 'mellivora';
+GRANT ALL PRIVILEGES ON mellivora.* TO 'mellivora'@'%';
+SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));
+
```
Comment (dbsqwerty) [inline db/db.sql:5]: are we going to have a command to add an admin user?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+WORKDIR /var/www/ctfx
+RUN composer install --no-dev --optimize-autoloader
+
```
Comment (dbsqwerty) [inline Dockerfile:21]: is there no need to copy the db.default.inc.php and config.default.inc.php to db.inc.php and config.inc.php respectively. Perhaps we could either make use of sed to write the config set in docker-compose or to get the env vars set in docker-compose to apply to the config files

File: db/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+# needed for intialization
+ENV MYSQL_ROOT_PASSWORD="seandontforgetmewhenyougetrichpls"
+
+COPY db.sql /docker-entrypoint-initdb.d/
+
+RUN ["/usr/local/bin/docker-entrypoint.sh", "mysqld", "--datadir", "/initialized-db"]
+
+FROM mysql:8.0
+
```
Comment (dbsqwerty) [inline db/Dockerfile:14]: same as the other Dockerfile. Is there a way to incorporate env configs in the setting up of the database as well as the setting up of the admin user account

File: db/db.sql (hunk 3 above)
Comment (dbsqwerty) [inline db/db.sql:5]: are we going to have a command/query to add an admin user?

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      - ./htdocs/:/var/www/ctfx/htdocs/
+    ports: 
+      - "8088:80"
+  database:
+    container_name: database
+    build:
+      context: ./db
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: "seandontforgetmewhenyougetrichpls"
```
Comment (dbsqwerty) [inline docker-compose.yaml:25]: as mentioned earlier, will we have the ability to set ENV configs when configuring ctfx

File: include/config/db.default.inc.php [hunk 7]
```diff
@@ -22,4 +22,4 @@
 Config::set('MELLIVORA_CONFIG_DB_PORT', 3306);
 Config::set('MELLIVORA_CONFIG_DB_NAME', 'mellivora');
 Config::set('MELLIVORA_CONFIG_DB_USER', 'mellivora');
-Config::set('MELLIVORA_CONFIG_DB_PASSWORD', '');
+Config::set('MELLIVORA_CONFIG_DB_PASSWORD', 'mellivora');
```
Comment (dbsqwerty) [inline include/config/db.default.inc.php:25]: hacky setting?

**Comment by dbsqwerty** [comment]
the markdown part looks fine actually.

File: include/config/db.default.inc.php (hunk 7 above)
Comment (Oceankoh) [reply include/config/db.default.inc.php:25]: i was thinking just create like a db.docker.inc.php which is copied to like db.inc.php in docker build

Comment (dbsqwerty) [reply include/config/db.default.inc.php:25]: could we instead use environment variables to set this?

**Comment by dbsqwerty** [comment]
I believe that should have resolved the bugs from an endless restart loop

**Comment by dbsqwerty** [comment]
Dockerisation LGTM.

**Review by dbsqwerty — APPROVED** [review]

**Pull request merged**